cirq/ops/parity_gates_test.py

```diff
@@ -0,0 +1,322 @@
+
+
+def test_xx_consistent():
+    cirq.testing.assert_eigen_gate_has_consistent_apply_unitary(cirq.XXPowGate)
+
+    cases = [cirq.XX,
+             cirq.XX**0.1,
+             cirq.XXPowGate(global_shift=0.1, exponent=0.2),
+             cirq.XXPowGate(global_shift=-0.5, exponent=0.4)]
+    for case in cases:
+        cirq.testing.assert_qasm_is_consistent_with_unitary(case)
+        cirq.testing.assert_decompose_is_consistent_with_unitary(case)
+        cirq.testing.assert_phase_by_is_consistent_with_unitary(case)
+
+
+def test_xx_diagrams():
+    a = cirq.NamedQubit('a')
+    b = cirq.NamedQubit('b')
+    circuit = cirq.Circuit.from_ops(
+        cirq.XX(a, b),
+        cirq.XX(a, b)**3,
+        cirq.XX(a, b)**0.5,
+    )
+    cirq.testing.assert_has_diagram(circuit, """
+a: ───XX───XX───XX───────
+      │    │    │
+b: ───XX───XX───XX^0.5───
+""")
+
+
+def test_yy_init():
+    assert cirq.YYPowGate(exponent=1).exponent == 1
+    v = cirq.YYPowGate(exponent=0.5)
+    assert v.exponent == 0.5
+
+
+def test_yy_eq():
+    eq = cirq.testing.EqualsTester()
+    eq.add_equality_group(cirq.YY,
+                          cirq.YYPowGate(),
+                          cirq.YYPowGate(exponent=1, global_shift=0),
+                          cirq.YYPowGate(exponent=3, global_shift=0))
+    eq.add_equality_group(cirq.YY**0.5, cirq.YY**2.5, cirq.YY**4.5)
+    eq.add_equality_group(cirq.YY**0.25, cirq.YY**2.25, cirq.YY**-1.75)
+
+    iYY = cirq.YYPowGate(global_shift=0.5)
+    eq.add_equality_group(iYY**0.5, iYY**4.5)
+    eq.add_equality_group(iYY**2.5, iYY**6.5)
+
+
+def test_yy_pow():
+    assert cirq.YY**0.5 != cirq.YY**-0.5
+    assert cirq.YY**-1 == cirq.YY
+    assert (cirq.YY**-1)**0.5 == cirq.YY**-0.5
+
+
+def test_yy_str():
+    assert str(cirq.YY) == 'YY'
+    assert str(cirq.YY**0.5) == 'YY**0.5'
+    assert str(cirq.YYPowGate(global_shift=0.1)) == 'YY'
+
+    iYY = cirq.YYPowGate(global_shift=0.5)
+    assert str(iYY) == 'YY'
+
+
+def test_yy_repr():
+    assert repr(cirq.YYPowGate()) == 'cirq.YY'
+    assert repr(cirq.YYPowGate(exponent=0.5)) == '(cirq.YY**0.5)'
+
+    cirq.testing.assert_equivalent_repr(cirq.YY)
+    cirq.testing.assert_equivalent_repr(cirq.YY**0.1)
+    cirq.testing.assert_equivalent_repr(cirq.YY**0.5)
+    cirq.testing.assert_equivalent_repr(cirq.YY**2)
+    cirq.testing.assert_equivalent_repr(cirq.YYPowGate(exponent=0.2,
+                                                       global_shift=0.3))
+
+
+def test_yy_matrix():
+    np.testing.assert_allclose(cirq.unitary(cirq.YY),
+                               np.array([[0, 0, 0, -1],
+                                         [0, 0, 1, 0],
+                                         [0, 1, 0, 0],
+                                         [-1, 0, 0, 0]]),
+                               atol=1e-8)
+    np.testing.assert_allclose(cirq.unitary(cirq.YY**2),
+                               np.eye(4),
+                               atol=1e-8)
+    c = np.cos(np.pi/4)
```

Comment:
See other comment above for better angles.

Reply:
Done.